src/main/java/io/cdap/plugin/gcp/bigquery/actions/BigQueryMergeActionConfig.java

```diff
@@ -0,0 +1,125 @@
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.etl.api.validation.InvalidConfigPropertyException;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryUtil;
+import io.cdap.plugin.gcp.common.GCPConfig;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Base class for Big Query Merge action configs.
+ */
+public class BigQueryMergeActionConfig extends GCPConfig {
+
+  public static final String NAME_TABLE_KEY = "relationTableKey";
+
+  @Macro
+  @Description("The dataset the table belongs to. A dataset is contained within a specific project. "
+    + "Datasets are top-level containers that are used to organize and control access to tables and views.")
+  private String sourceDataset;
+
+  @Macro
+  @Description("The table to read from. A table contains individual records organized in rows. "
+    + "Each record is composed of columns (also called fields). "
+    + "Every table is defined by a schema that describes the column names, data types, and other information.")
+  private String sourceTable;
+
+  @Macro
+  @Description("The dataset to write to. A dataset is contained within a specific project. "
+    + "Datasets are top-level containers that are used to organize and control access to tables and views.")
+  private String destinationDataset;
+
+  @Macro
+  @Description("The table to write to. A table contains individual records organized in rows. "
+    + "Each record is composed of columns (also called fields). "
+    + "Every table is defined by a schema that describes the column names, data types, and other information.")
+  private String destinationTable;
+
+  @Macro
+  @Description("Whether to modify the BigQuery destination table schema if it differs from the source table schema.")
+  private boolean allowSchemaRelaxation;
+
+  @Macro
+  @Description("List of fields that determines relation between source table and destination table.")
+  private String relationTableKey;
+
+  public void validate() {
+    validateTableExists(sourceDataset, sourceTable);
+    validateTableExists(destinationDataset, destinationTable);
+    validateOperationProperties();
+  }
+
+  private void validateTableExists(String datasetName, String tableName) {
+    Table table = BigQueryUtil.getBigQueryTable(project, datasetName, tableName, serviceFilePath);
+
+    if (table == null) {
+      throw new IllegalStateException(String.format("BigQuery table '%s:%s.%s' does not exist", project, sourceDataset,
+                                                    sourceTable));
+    }
+  }
+
+  private void validateOperationProperties() {
+    Table table = BigQueryUtil.getBigQueryTable(project, sourceDataset, sourceTable, serviceFilePath);
+
+    FieldList sourceFields = table.getDefinition().getSchema().getFields();
+    List<String> fields = sourceFields.stream().map(Field::getName).collect(Collectors.toList());
```

Comment:
as for me, it looks better when we wrap chain calls so we have 1 method call per line